go-controller/pkg/util/net_linux.go

```diff
@@ -170,27 +171,36 @@ func LinkNeighExists(link netlink.Link, neighIP net.IP, neighMAC net.HardwareAdd
 	return false, nil
 }
 
-func DeleteConntrack(ip string, port int32) {
+func DeleteConntrack(ip string, port int32, protocol kapi.Protocol) {
 	ipAddress := net.ParseIP(ip)
 	if ipAddress == nil {
 		klog.Errorf("Value \"%s\" passed to DeleteConntrack is not an IP address", ipAddress)
 		return
 	}
 
 	filter := &netlink.ConntrackFilter{}
-	// 17 = UDP protocol
-	err := filter.AddProtocol(17)
-	if err != nil {
-		klog.Warningf("Could not add Protocol UDP to conntrack filter %v", err)
-		return
+	if protocol == kapi.ProtocolUDP {
+		// 17 = UDP protocol
+		err := filter.AddProtocol(17)
+		if err != nil {
+			klog.Warningf("Could not add Protocol UDP to conntrack filter %v", err)
+			return
+		}
+	} else if protocol == kapi.ProtocolSCTP {
+		// 132 = SCTP protocol
+		err := filter.AddProtocol(132)
+		if err != nil {
+			klog.Warningf("Could not add Protocol UDP to conntrack filter %v", err)
+			return
+		}
 	}
 	if port > 0 {
-		err = filter.AddPort(netlink.ConntrackOrigDstPort, uint16(port))
+		err := filter.AddPort(netlink.ConntrackOrigDstPort, uint16(port))
 		if err != nil {
 			klog.Warningf("Could not add port %d to conntrack filter: %v", port, err)
```

Comment:
@JacobTanenbaum do we need a `return` here?

Reply:
done